src/rviz/default_plugin/tf_display.cpp

```diff
@@ -372,22 +377,34 @@ void TFDisplay::updateFrames()
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
 
-  context_->getTFClient()->getFrameStrings(frames);
+context_->getTFClient()->getFrameStrings(frames);
 
 #ifndef _WIN32
 #pragma GCC diagnostic pop
 #endif
   std::sort(frames.begin(), frames.end());
 
   S_FrameInfo current_frames;
+  std::string whitelist_regex_string = filter_property_->getStdString();
+  std::string blacklist_regex_string = filter_blacklist_property_->getStdString();
+  auto whitelist_regex = std::regex(whitelist_regex_string);
+  auto blacklist_regex = std::regex(blacklist_regex_string);
 
   {
     for (const std::string& frame : frames)
     {
       if (frame.empty())
-      {
         continue;
+
+      try
+      {
+        if (!std::regex_match(frame, whitelist_regex))
+          continue;
+        if (!blacklist_regex_string.empty())
+          if (std::regex_match(frame, blacklist_regex))
```

Comment:
Prefer logical AND over nested IF:
```suggestion
        if (!blacklist_regex_string.empty()) && std::regex_match(frame, blacklist_regex))
```